vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/codec/json/JsonObjectCodec.java

```diff
@@ -106,6 +149,14 @@ protected Object readObjectId(BsonReader reader, DecoderContext ctx) {
 
   @Override
   protected Object readDateTime(BsonReader reader, DecoderContext ctx) {
```

Comment:
Acutally that would work.  The behavior in question lies with the Mongo DB JSON parser.  The parser used by the Java driver does not enforce this split personallity.  See `org.bson.json.JsonReader.visitDateTimeConstructor`.  However if that's your goal then you may have another issue which is the expected format.  I can put together some sample code to demonstrate but it will have to wait a couple of hours (have some household stuff to do first).

However, I have to say that having working with the Java driver for a few years now I've never done this -- primarilary because it supports the native Date objects.  Not that a sample size of 1 means much, but I think the primary goal has to be reasonable usage within Vert.x and exposing all this mess directly doesn't meet that goal IMO.